### PR TITLE
Give human readable errors when schedule name is invalid.

### LINKED ISSUE #320: Improve the error message for invalid cron expressions
Currently if a user attempts to create a schedule name that has characters (like capital letters) that are not accepted by kubernetes, our exit message is the default returned from fabric8 which is basically the regex expression ` ^a-z0-9([a-z0-9]*[a-z0-9])?$`.   We need to return a friendlier message.   More information is found here: https://github.com/spring-cloud/spring-cloud-dataflow/issues/3443

### PR DESCRIPTION
**Pull request by cppwfs**
resolves #320

### CODE REVIEW

File: src/main/java/org/springframework/cloud/deployer/scheduler/spi/kubernetes/KubernetesScheduler.java [hunk 1]
```diff
@@ -81,6 +85,16 @@ public void schedule(ScheduleRequest scheduleRequest) {
 		}
 	}
 
+	public void validateScheduleName(ScheduleRequest request) {
+		if(request.getScheduleName() == null) {
+			throw new IllegalArgumentException("The name for the schedule request is null");
```
Comment (chrisjs) [inline src/main/java/org/springframework/cloud/deployer/scheduler/spi/kubernetes/KubernetesScheduler.java]: why throw IllegalArgumentException in both places of this method vs CreateScheduleException? we throw that when the cron expression isnt valid as well

Comment (cppwfs) [reply src/main/java/org/springframework/cloud/deployer/scheduler/spi/kubernetes/KubernetesScheduler.java]: Good Catch

Comment (cppwfs) [reply src/main/java/org/springframework/cloud/deployer/scheduler/spi/kubernetes/KubernetesScheduler.java]: As it pertains to the cron expression, the exceptions we get back from k8's are pretty good.  
i.e.
```
Failed to create schedule CronJob.batch "bbb" is invalid: spec.schedule: Invalid value: "*/1 * * R C": Failed to parse int from R: strconv.Atoi: parsing "R": invalid syntax
``` 
when I tried to enter a cron expression of `*/1 * * R C`

**Comment by chrisjs** [comment]
merged in 5304c785b6d1139c268441e2ec48cc65d3df6d76

**Pull request closed without merging**